manifests/params.pp

```diff
@@ -23,6 +23,11 @@
   $qpid_client_cert = "${pulp_pki_dir}/qpid/client.crt"
   $qpid_client_ca_cert = "${pulp_pki_dir}/qpid/ca.crt"
 
+  # Generate and cache the password on the master once
+  # In multi-puppetmaster setups, the user should specify their own
+  $ca_key_password = extlib::cache_data('foreman_cache_data', 'ca_key_password', extlib::random_password(24))
+  $nss_db_password = extlib::cache_data('foreman_cache_data', 'certs-nss-db-password', extlib::random_password(32))
```

Comment:
What's the reason to move the NSS DB pass to `params.pp`? Looks like we can continue using it in `certs::ssltools:nssdb`. You do need to set it via Hiera, but you don't introduce it as a top level parameter anyway.

Reply:
This was just following a pattern. Most of the other modules I looked at had this kind of thing done in `params.pp`, so I just did it there as well. I can move it out to the class itself if that makes more sense.

Reply:
Ideally we would get rid of `params.pp`. In our installer we use Kafo, which can't deal with function calls in parameters (default parsing breaks). Now I do see that `ca_key_password` already was doing that in `init.pp` so now I wonder if that was broken all along.

Reply:
Ah, I missed something in my review: it's in the body now, not the actual class parameter. That's why it's not a problem today.